src/content/en/updates/2018/06/audio-worklet-design-pattern.md

```diff
@@ -0,0 +1,414 @@
+project_path: /web/_project.yaml
+book_path: /web/updates/_book.yaml
+
+{# wf_published_on: 2018-06-18 #}
+{# wf_updated_on: 2018-06-26 #}
+{# wf_featured_image: /web/updates/images/generic/webaudio.png #}
+{# wf_tags: chrome68,webaudio,worklet #}
+{# wf_featured_snippet: Advanced design patterns to unlock AudioWorklet's fullest power with WebAssembly and SharedArrayBuffer. #}
+{# wf_blink_components: Blink>WebAudio #}
+
+<style>
+  .diagram { max-width: 90%; display: block; margin: 0 auto; padding: 12px; }
+  figcaption { font-size: 14px; text-align: center; padding: 8px; }
+</style>
+
+# AudioWorklet Design Pattern {: .page-title }
+
+{% include "web/_shared/contributors/hongchanchoi.html" %}
+
+The [previous article](/web/updates/2017/12/audio-worklet)
+on the AudioWorklet detailed the basic concepts and its usage. Since its launch
+in Chrome 66 there have been many requests for more examples of how it can be
+used in actual applications. The AudioWorklet unlocks the full potential of
+WebAudio, but taking advantage of it can be challenging because it requires
+understanding concurrent programming wrapped with several JS APIs. Even for
+developers who are familiar with the WebAudio, integrating the AudioWorklet
+with other APIs (e.g. WebAssembly) can be difficult.
+
+This article will give the reader a better understanding of how to use the
+AudioWorklet in real-world settings and to offer tips to draw on its fullest
+power. Be sure to check out
+[code examples and live demos](https://googlechromelabs.github.io/web-audio-samples/audio-worklet#design-pattern)
+as well!
+
+## Recap: AudioWorklet
+
+Before diving in, let's quickly recap terms and facts around the AudioWorklet
+system which is previously introduced in
+[this post](/web/updates/2017/12/audio-worklet).
+
+  - [BaseAudioContext](https://developer.mozilla.org/en-US/docs/Web/API/BaseAudioContext):
+    Web Audio API's primary object.
```

Comment:
Maybe be consistent and write "WebAudio" or "Web Audio" (no space or space)?

Reply:
I think they are slightly different. IMO WebAudio is a generic term for things written in Web Audio API. OTOH Web Audio API is literally a name of the API. Perhaps it's just me, but I've been using it that way.